src/Printed/Bundle/Queue/Queue/AbstractQueueConsumer.php

```diff
@@ -45,71 +48,46 @@ abstract class AbstractQueueConsumer implements ConsumerInterface, ServiceSubscr
+     * this abstract class' protected methods to make changes to it.
      */
-    protected $locator;
+    protected ?QueueTaskInterface $task = null;
 
-    /** @var ServiceContainerParameters */
-    protected $containerParameters;
+    private QueueBundleOptions $queueBundleOptions;
 
-    /** @var AMQPMessage */
-    protected $message;
+    private NewDeploymentsDetector $newDeploymentsDetector;
 
     /**
-     * This entity might be managed by a separate entity manager. Treat it as read-only in your subclass, and use
-     * this abstract class' protected methods to make changes to it.
-     *
-     * @var QueueTaskInterface
+     * @see NewDeploymentsDetector::getCurrentDeploymentStamp()
      */
-    protected $task;
+    private string $startUpDeploymentStamp;
 
     /**
-     * @var EntityManager The (potentially dedicated) entity manager to be used by this abstract class only. Part of
+     * @param EntityManagerInterface $em This is always the application's entity manager
+     * @param ValidatorInterface $validator
+     * @param LoggerInterface $logger
+     * @param ContainerInterface $locator Services requested by ::getSubscribedServices() are available in this container.
+     * @param ServiceContainerParameters $containerParameters
+     * @param EntityManagerInterface|null $internalQueueConsumerEntityManager The (potentially dedicated) entity manager to be used by this abstract class only. Part of
      *  preventing the "closed" application's entity manager from influencing the ability of this abstract class to
      *  track failed queue tasks correctly.
+     * @param DateTime $startUpDateTime The time the consumer were constructed.
+     *
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
      */
-    private $internalQueueConsumerEntityManager;
-
-    /** @var QueueBundleOptions */
-    private $queueBundleOptions;
-
-    /** @var NewDeploymentsDetector */
-    private $newDeploymentsDetector;
-
-    /**
-     * @var string
-     * @see NewDeploymentsDetector::getCurrentDeploymentStamp()
-     */
-    private $startUpDeploymentStamp;
-
-    /** @var \DateTime The time the consumer were constructed. */
-    private $startUpDateTime;
-
     public function __construct(
-        EntityManagerInterface $em,
-        ValidatorInterface $validator,
-        LoggerInterface $logger,
-        ContainerInterface $locator,
-        ServiceContainerParameters $containerParameters,
-        ?EntityManagerInterface $internalQueueConsumerEntityManager = null
+       protected EntityManagerInterface $em,
+       protected ValidatorInterface $validator,
+       protected LoggerInterface $logger,
+       protected ContainerInterface $locator,
+       protected ServiceContainerParameters $containerParameters,
+       protected ?EntityManagerInterface $internalQueueConsumerEntityManager = null,
+       private readonly DateTime $startUpDateTime = new DateTime(),
```

Comment:
Please revert this argument to be initialised in the constructor body, and the field itself being declared in the previous way. This field shouldn't be initialise-able by a constructor argument.